sqlitecluster/SQLiteClusterMessenger.h

```diff
@@ -50,6 +71,5 @@ class SQLiteClusterMessenger {
     atomic_flag _shutdownSet = ATOMIC_FLAG_INIT;
 
     // For managing many connections to leader, we have a socket pool.
-    mutex _socketPoolMutex;
-    unique_ptr<SSocketPool> _socketPool;
+    unique_ptr<SMultiSocketPool> _socketPool;
```

Comment:
This doesn't really need to be wrapped in a `unique_ptr` if it's always onwed by the same object and has a very specific lifetime (same as the owning object). you could just use `SMultiSocketPool _socketPool` and it'd be simpler.